### PR TITLE
Remove unit snapshots

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING CI && PENDING REVIEW

### What does this PR accomplish?
- Removes unit tests snapshots and all related files (this was agreed upon offline among the core FE team members)
- All tests should still pass in CI

### Question
- Do we plan to use `react-test-renderer` or do we want to remove it as well? It is currently not used anywhere else in the app.

### CODE REVIEW

**Review by alxnddr — APPROVED** [review]
Thanks for handling this 🙏

**Review by kulyk — APPROVED** [review]
Thank you!

**Review by gusaiani — COMMENTED** [review]
This is amazing Nemanja, thank you for this. I should save us hundreds of person-hours per year as soon as we are on these changes.

Having chatted with @ariya about pointing PRs to `release` or `master` branches, I am uncertain if we'd be better off pointing this to `master`.

Maybe Ariya can share his pov?

File: package.json [hunk 1]
```diff
@@ -168,7 +168,6 @@
     "test": "yarn test-unit && yarn test-timezones && yarn test-cypress",
     "test-unit": "yarn build-quick:cljs && jest --maxWorkers=2 --config jest.unit.conf.json",
     "test-unit-watch": "yarn test-unit --watch",
-    "test-unit-update-snapshot": "yarn test-unit --updateSnapshot",
```
Comment (gusaiani) [inline package.json:171]: 🎉

File: frontend/test/metabase/internal/components.unit.spec.js [hunk 2]
```diff
@@ -1,19 +0,0 @@
-import renderer from "react-test-renderer";
-
-import components from "metabase/internal/lib/components-node";
-
-// generates a snapshot test for every example in every component's `.info.js`
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/internal/components.unit.spec.js:5]: Thanks for bringing this to my attention offline. Removed it:
https://github.com/metabase/metabase/pull/16321/commits/a185e768db9aaeb243c295bb2dab773ccf4a5304

File: frontend/test/metabase/components/Calendar.unit.spec.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import React from "react";
-import renderer from "react-test-renderer";
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/components/Calendar.unit.spec.js:2]: Done: https://github.com/metabase/metabase/pull/16321/commits/5febe37739a7f8359aa42d0916113ab397b0e2c0

**Comment by ariya** [comment]
> Maybe Ariya can share his pov?

@nemanjaglumac is already correct. Doing otherwise will stage a future merge conflict.

**Pull request merged**